src/tools/miri/cargo-miri/src/setup.rs

```diff
@@ -136,7 +136,9 @@ pub fn setup(
     // But keep the overflow checks, they are cheap. This completely overwrites flags
     // the user might have set, which is consistent with normal `cargo build` that does
     // not apply `RUSTFLAGS` to the sysroot either.
-    let rustflags = &["-Cdebug-assertions=off", "-Coverflow-checks=on"];
+    // We also need to set `force-unstable-if-unmarked` to ensure the sysroot builds.
+    let rustflags =
+        &["-Cdebug-assertions=off", "-Coverflow-checks=on", "-Zforce-unstable-if-unmarked"];
```

Comment:
The recursive const stability logic relies on `-Zforce-unstable-if-unmarked`, so hashbrown needs to be built with that flag to ensure std can expose hashbrown `const fn` on stable.

Now I wonder if this won't be a problem for `-Zbuild-std`... does that set `-Zforce-unstable-if-unmarked`?

Reply:
Seems like the answer is yes, cargo does set that flag:

https://github.com/rust-lang/cargo/blob/668d82e3f0461ebbcb8eb2d86a8150be494a9427/src/cargo/core/compiler/mod.rs#L1214-L1223

Reply:
Turns out Miri also already sets that flag, the actual issue was setting in in boostrap.